### PR TITLE
add colours for man pages

### PR DESCRIPTION
**Pull request by benswift**
Another PR to add colors for a mode - in this case `Man-mode`.

Mostly, this is motivated by the fact that I switched to spacemacs-theme from monokai, and there are a few mode-specific colours that I miss. I admit that I'm not a colours-knowing-expert guy, so I'm totally fine if you want to tweak the colors that I've chosen - the main thing I like is that there are *some* pretty colors.

In future, do you want extra mode-specific colors done this way? I'm aware that not everyone likes it to look the way I do. I'd be happy to just set up these colors in my init file, except the way that the theme (and emacs themes in general, it seems) do the name -> colour mapping is a bunch of `let` bindings in a bigol' macro, so it's tricky to use those mappings elsewhere in one's init file. Still, I'm open to suggestions about how to walk the line between having colours where *I* like them, but not imposing them on everyone.

Sorry about the long rant :)

### CODE REVIEW

**Comment by nashamri** [comment]
Awesome :+1: keep these PRs coming :smile: 

One tiny thing that needs to be changed for this PR is changing the `:weight bold` to `:inhert bold` because of this https://github.com/nashamri/spacemacs-theme/issues/37

For future PRs, I think you only need to pick the logical color variable (see the table here: https://github.com/nashamri/spacemacs-theme#override-themes-colors) that best matches what you are trying to theme. My goal is to have this layer of abstraction for all the supported modes and if that's done right then the _per user_ customization can be easily achieved by simple overriding these variables like so:
```elisp
  (custom-set-variables '(spacemacs-theme-custom-colors
                          '((act1 . "#ff0000")
                            (act2 . "#0000ff")
                            (base . "#ffffff"))))
```

**Comment by benswift** [comment]
> One tiny thing that needs to be changed for this PR is changing the
> `:weight bold` to `:inhert bold` because of this
> https://github.com/nashamri/spacemacs-theme/issues/37

No worries, want me to make the change, or do you want to do it?

> For future PRs, I think you only need to pick the logical color
> variable (see the table here:
> https://github.com/nashamri/spacemacs-theme#override-themes-colors)
> that best matches what you are trying to theme. My goal is to have
> this layer of abstraction for all the supported modes and if that's
> done right then the _per user_ customization can be easily achieved by
> simple overriding these variables like so:

Cool, so you mean use these colours?

    aqua
    aqua-bg
    green
    green-bg
    green-bg-s
    cyan
    red
    red-bg
    red-bg-s
    blue
    blue-bg
    violet
    yellow
    yellow-bg

**Comment by nashamri** [comment]
> No worries, want me to make the change, or do you want to do it?

Yeah if you can amend that commit, that would be grean and I think it'll be quicker, or I can do it after I merge it. It's not a big deal :smile: 

> Cool, so you mean use these colours?
> 
>     aqua
>     aqua-bg
>     green
>     green-bg
>     green-bg-s
>     cyan
>     red
>     red-bg
>     red-bg-s
>     blue
>     blue-bg
>     violet
>     yellow
>     yellow-bg
> 

Github always messes the subsections links :joy: 
The color variables you mentioned are okay to be used, but they are meant to be used less often than these:

| var        | role                                                                                              |
|------------|---------------------------------------------------------------------------------------------------|
| act1       | One of mode-line's active colors.                                                                 |
| act2       | The other active color of mode-line.                                                              |
| base       | The basic color of normal text.                                                                   |
| base-dim   | A dimmer version of the normal text color.                                                        |
| bg1        | The background color.                                                                             |
| bg2        | A darker background color. Used to highlight current line.                                        |
| bg3        | Yet another darker shade of the background color.                                                 |
| bg4        | The darkest background color.                                                                     |
| border     | A border line color. Used in mode-line borders.                                                   |
| cblk       | A code block color. Used in org's code blocks.                                                    |
| cblk-bg    | The background color of a code block.                                                             |
| cblk-ln    | A code block header line.                                                                         |
| cblk-ln-bg | The background of a code block header line.                                                       |
| cursor     | The cursor/point color.                                                                           |
| const      | A constant.                                                                                       |
| comment    | A comment.                                                                                        |
| comment-bg | The background color of a comment. To disable this, `customize` `spacemacs-theme-comment-bg`.     |
| comp       | A complementary color.                                                                            |
| err        | errors.                                                                                           |
| func       | functions.                                                                                        |
| head1      | Level 1 of a heading. Used in org's headings.                                                     |
| head1-bg   | The background of level 2 headings. To disable this, `customize` `spacemacs-theme-org-highlight`. |
| head2      | Level 2 headings.                                                                                 |
| head2-bg   | Level 2 headings background.                                                                      |
| head3      | Level 3 headings.                                                                                 |
| head3-bg   | Level 3 headings background.                                                                      |
| head4      | Level 4 headings.                                                                                 |
| head4-bg   | Level 4 headings background.                                                                      |
| highlight  | A highlighted area.                                                                               |
| keyword    | A keyword or a builtin color.                                                                     |
| lnum       | Line numbers.                                                                                     |
| mat        | A matched color. Used in matching parens, brackets and tags.                                      |
| meta       | A meta line. Used in org's meta line.                                                             |
| str        | A string.                                                                                         |
| suc        | To indicate success. Opposite of error.                                                           |
| ttip       | Tooltip color.                                                                                    |
| ttip-sl    | Tooltip selection color.                                                                          |
| ttip-bg    | Tooltip background color.                                                                         |
| type       | A type color.                                                                                     |
| var        | A variable color.                                                                                 |
| war        | A warning color.                                                                                  |

It might be a bit confusing what's the difference between the two sets? Let me try to explain why I did it this way by an example:

Let's take the variable `mat` which indicates a **matching** of some sort. I've used it for smartparens to show the matching paren. So, if you used the dark variant, it'll be a green-ish color and if you used the light variant it will a red-ish color. So, using the variables in the table do not guarantee how the color will look like across variants.

Now if we take the `red` color variable, I've used where I needed a guarantee that this color will be red across variants. For instance, in `ediff` or in `magit` where something is being *deleted*. Deletion is usually expressed by a red color and addition is expressed by a green color. So here the color has a bit of a semantic meaning and I wanted to keep this meaning across variants.

Sorry for making this a bit philosophical lol, it might not be the best way to do things and I'm not sure I could come up with what `aqua` means semantically :joy: it might only make sense in the red and green cases.

tl;dr
Use the colors from the table I mentioned above and unless you need to guarantee something to be red or green or whatever across the variants, then use the colors you mentioned.

Hope this helps :smile:

**Comment by benswift** [comment]
> Yeah if you can amend that commit, that would be grean and I think it'll be quicker, or I can do it after I merge it. It's not a big deal :smile: 

Cool, I pushed up an update.

> It might be a bit confusing what's the difference between the two sets? Let me try to explain why I did it this way by an example:
>
> Let's take the variable `mat` which indicates a **matching** of some sort. I've used it for smartparens to show the matching paren. So, if you used the dark variant, it'll be a green-ish color and if you used the light variant it will a red-ish color. So, using the variables in the table do not guarantee how the color will look like across variants.
>
> Now if we take the `red` color variable, I've used where I needed a guarantee that this color will be red across variants. For instance, in `ediff` or in `magit` where something is being *deleted*. Deletion is usually expressed by a red color and addition is expressed by a green color. So here the color has a bit of a semantic meaning and I wanted to keep this meaning across variants.
>
> Sorry for making this a bit philosophical lol, it might not be the best way to do things and I'm not sure I could come up with what `aqua` means semantically :joy: it might only make sense in the red and green cases.
>
> tl;dr
> Use the colors from the table I mentioned above and unless you need to guarantee something to be red or green or whatever across the variants, then use the colors you mentioned.

Ok, cool. I guess I tried to do that with the previous PR, using
head{1,2} rather than blue or green, but with the update I've tried to
"semantically match" the faces to the appropriate values in the list.
Like I said, though, if you think there are better mappings, then I'm
open to them.

> Hope this helps :smile: 

Great, yep, it does :)

**Comment by nashamri** [comment]
Brilliant! thanks @benswift :+1:
